capstone/feed-fetching/test_get_vt_feed.py

```diff
@@ -0,0 +1,48 @@
+import bz2
+import io
+import os
+import pytest
+import requests
+from datetime import datetime
+from unittest.mock import patch, Mock
+
+
+class MockResponse:
+    def __init__(self, status_code=200, text='', content=b''):
+        self.status_code = status_code
+        self.text = text
+        self.content = content
+
+
+@patch('os.environ.get', return_value='test-key')
+@patch('requests.get', return_value=MockResponse(status_code=200, content=b'vt-feed'))
+@patch('bz2.open', return_value=io.BytesIO(b'{ "attributes": { "attr1": "val1", "attr2": "val2" } }'))
+def test_get_feed(bz2_open, requests_get, os_environ):
+    import get_vt_feed
+    result = list(get_vt_feed.get_feed(0))
+
+    os.environ.get.assert_called_once_with('VTKEY')
+    filename = datetime.utcnow().strftime('%Y%m%d%H%M')
+    requests.get.assert_called_once_with('https://www.virustotal.com/api/v3/feeds/files/{}'.format(filename),
+                                         headers={'X-Apikey': 'test-key'})
+    bz2.open.assert_called_once()
+    assert result == [{'attr1': 'val1', 'attr2': 'val2'}]
+
+
+def test_prepare_doc():
+    import get_vt_feed
```

Comment:
why are you importing this at function level?

Reply:
Because I need to patch global variables (`apikey`) before importing

Reply:
You can store the original's value in a class variable, set the value you need for the test and then reset its original value. You can do that by using setUp and tearDown (https://docs.python.org/3/library/unittest.html#setupclass-and-teardownclass)